src/components/Markdown.react.js

```diff
@@ -1,35 +1,76 @@
-import React from 'react';
+import React, {Component} from 'react';
 import PropTypes from 'prop-types';
 import {omit, propOr, type} from 'ramda';
 import Markdown from 'react-markdown';
+import './css/highlight.css';
 
 // eslint-disable-next-line valid-jsdoc
 /**
  * A component that renders Markdown text as specified by the
  * GitHub Markdown spec. These component uses
  * [react-markdown](https://rexxars.github.io/react-markdown/) under the hood.
  */
-function DashMarkdown(props) {
-    if (type(props.children) === 'Array') {
-        props.children = props.children.join('\n');
+class DashMarkdown extends Component {
+    componentDidMount() {
+        this.highlightCode();
     }
 
-    return (
-        <div
-            id={props.id}
-            data-dash-is-loading={
-                (props.loading_state && props.loading_state.is_loading) ||
-                undefined
-            }
-            {...propOr({}, 'containerProps', props)}
-        >
-            <Markdown
-                source={props.children}
-                escapeHtml={!props.dangerously_allow_html}
-                {...omit(['containerProps'], props)}
-            />
-        </div>
-    );
+    componentDidUpdate() {
+        this.highlightCode();
+    }
+
+    highlightCode() {
+        if (!window.hljs) {
+            // skip highlighting if highlight.js isn't found
+            return;
+        }
+        const nodes = document.querySelectorAll('pre code');
```

Comment:
ooh that's too greedy (and potentially slow) - can we restrict it to this component?

Reply:
Yeah, agreed. `querySelectorAll` is [really slow](https://jsperf.com/queryselectorall-vs-getelementsbytagname/20). I changed it to use a React ref callback combined with `getElementsByTagName`.